backend/jobsearch/src/main/kotlin/com/jobsearch/service/VacancyService.kt

```diff
@@ -1,87 +1,111 @@
 package com.jobsearch.service
 
-import com.jobsearch.dto.VacancyDto
+import com.jobsearch.dto.VacancyRequestDTO
+import com.jobsearch.dto.VacancyResponseDTO
 import com.jobsearch.entity.Vacancy
+import com.jobsearch.exception.ForbiddenException
+import com.jobsearch.exception.NotFoundException
 import com.jobsearch.repository.VacancyRepository
 import jakarta.transaction.Transactional
 import org.springframework.stereotype.Service
+import java.util.*
+import kotlin.jvm.optionals.getOrElse
 
 @Service
 class VacancyService(
     val vacancyRepository: VacancyRepository,
-    val jobFamilyService: JobFamilyService
+    val jobFamilyService: JobFamilyService,
+    val userService: UserService
 ) {
-    fun createVacancy(vacancyDto: VacancyDto): VacancyDto {
-        val selectedJobFamily = jobFamilyService.findByJobFamilyId(vacancyDto.jobFamilyId!!)
-            .orElseThrow { NoSuchElementException("No vacancy found with id ${vacancyDto.jobFamilyId}") }
+    @Transactional
```

Comment:
We could skip using `@Transactional` for the get operation when we just retrieve some data and do not modify anything